src/app/modules/lti/pages/lti/lti.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { forkJoin, Observable, of } from 'rxjs';
+import { catchError, map, mapTo, shareReplay, switchMap } from 'rxjs/operators';
+import { GetItemChildrenService, ItemChild } from 'src/app/modules/item/http-services/get-item-children.service';
+import { GetItemPathService } from 'src/app/modules/item/http-services/get-item-path.service';
+import { GetResultsService, Result } from 'src/app/modules/item/http-services/get-results.service';
+import { bestAttemptFromResults, defaultAttemptId } from 'src/app/shared/helpers/attempts';
+import { errorIsHTTPForbidden } from 'src/app/shared/helpers/errors';
+import { errorState } from 'src/app/shared/helpers/state';
+import { ResultActionsService } from 'src/app/shared/http-services/result-actions.service';
+import { mapToFetchState } from 'src/app/shared/operators/state';
+import { fullItemRoute } from 'src/app/shared/routing/item-route';
+import { ItemRouter } from 'src/app/shared/routing/item-router';
+import { LayoutService } from 'src/app/shared/services/layout.service';
+
+const explicitEntryWithNoResultError = new Error('no result');
+const noChildError = new Error('no first child');
+const noContentIdError = new Error('no content id');
+type LTIError = 'no content id' | 'fetch error' | 'explicit entry with no result' | 'no child';
```

Comment:
Remark (not to be changed necessarily): usually, in this project, string union types are done in snake case... the risk I see with plain text is to have later a developer use them in strings. Actually we just use union types instead of enums is that enums are boring to use in templates... btw, my latest way to handle this was to do enum with string values (for instance `TypeFilter` or `GroupManagership` or `Mode`).
Also, there is a similarity/duplication between `'no first child'` of the error and `'no child'` ... it may be misleading...  as `'no child'`  is in both the enum and the error.

Reply:
fixed